filters/united-states/washington/seattle/public-art-json-to-json.js

```diff
@@ -1,50 +1,109 @@
-const filter = function (data, std_lib, stringify, skip_errors) {
+const filter = function (data, std_lib, schema, validator, stringify) {
+    // check for standard library and pull out required functions
+    if (!std_lib) {
+        throw "standard library not provided";
+    }
+    let add_required = std_lib.get("add_required");
+    let add_if_not_null = std_lib.get("add_if_not_null")
+    let remove_if_null = std_lib.get("remove_if_null");
+    let remove_if_empty = std_lib.get("remove_if_empty");
+    let validate_params = std_lib.get("validate_params");
+    let create_dates_template = std_lib.get("create_dates_template");
+    let remove_null_date_fields = std_lib.get("remove_null_date_fields");
+
+    // validate parameters object
+    schema = validate_params(schema, validator);
+
     if (typeof data === 'string' || data instanceof String) {
         data = JSON.parse(data);
     }
-    
+
+    // define new data and errors array
+    let total_entries = 0;
     let new_data = [];
+    let errors = [];
 
     data.map(d => {
         let item = {};
-        item.name = d.title;
-        if (item.name === undefined) {
-            console.log(`Data name not found for art with url ${d.url}`);
+        let skip = false;
+        total_entries++;
+
+        // add name field
+        if (!add_required(item, "name", d, d.title, errors)) {
+            skip = true;
+        }
+
+        // add coordinate field
+        let coordinates = {};
+        if (!add_required(coordinates, "longitude", d, parseFloat(d.longitude), errors)) {
+            skip = true;
         }
-        let coordinates = { longitude: d.latitude, latitude: d.longitude };
-        if (coordinates.longitude === undefined || coordinates.latitude === undefined) {
-            console.log(`Data coordinates not found for art with url ${d.url}`);
+        if (!add_required(coordinates, "latitude", d, parseFloat(d.latitude), errors)) {
+            skip = true;
         }
         item.coordinates = coordinates;
 
-        // for optional not currently standardized fields.
+        // add artist field
+        if (d.artist_first_name || d.artist_last_name) {
```

Comment:
you could accidentally add in a null artist first or last with this, may be better to use && with another else if condition